dbms/src/Storages/Page/V3/Universal/UniversalPageStorageService.cpp

```diff
@@ -26,19 +36,100 @@ UniversalPageStorageServicePtr UniversalPageStorageService::create(
     auto service = UniversalPageStorageServicePtr(new UniversalPageStorageService(context));
     service->uni_page_storage = UniversalPageStorage::create(name, delegator, config, context.getFileProvider());
     service->uni_page_storage->restore();
-    service->gc_handle = context.getBackgroundPool().addTask(
+    auto & bkg_pool = context.getBackgroundPool();
+
+    if (S3::ClientFactory::instance().isEnabled())
+    {
+        // Only upload checkpoint when S3 is enabled
+        service->remote_checkpoint_handle = bkg_pool.addTask(
+            [service] {
+                return service->uploadCheckpoint();
+            },
+            /*multi*/ false,
+            /*interval_ms*/ 5 * 60 * 1000);
```

Comment:
TODO: make it configurable

Reply:
5 minutes is too long. The FAP code path is already timeout. How about use 10 seconds as a start point for the ease of test. Let's find a suitable default value later.

Reply:
Add a config in `Settings`, and default to 60 seconds